harvard-lts/drs-import-management-service
# ETD-297: Reinstate JWT Keys in DIMS and add ETD keys

**Reinstate JWT Keys in DIMS and add ETD keys**
* * *

**GitHub Issue**: https://at-harvard.atlassian.net/browse/ETD-297

# What does this Pull Request do?
adds example etd keys and restores jwt token validation

# How should this be tested?

1.  `cp .env.example .env`
2.  in the .env, set `JWT_KEYS `to `{"testKidDataverse":{"iss":"{{ dataverse_jwt_issuer }}","public_key_path":"/home/dimsuser/test/functional/test_dataverse_public.key","application_name":"Dataverse"}, "testKidEPADD":{"iss":"ePADD","public_key_path":"/home/dimsuser/test/functional/test_epadd_public.key","application_name":"ePADD"}, "testKidETD":{"iss":"{{ etd_jwt_issuer }}","public_key_path":"/home/dimsuser/test/functional/test_etd_public.key","application_name":"ETD"}}`
3. 
```
cd  test/functional/ 
cp test_etd_public.key.example test_etd_public.key  ; cp test_etd_private.key.example test_etd_private.key
cp test_epadd_public.key.example test_epadd_public.key  ; cp test_epadd_private.key.example test_epadd_private.key
cp test_dataverse_public.key.example test_dataverse_public.key  ; cp test_etd_dataverse_private.key.example test_dataverse_private.key
```
4.  `docker-compose -f docker-compose-local.yml up --build --force-recreate`
5.  `docker exec -it dims bash`
6.  `pytest test/functional/test_flask_app.py`   all tests (2) should pass. you will see 12 deprecation warnings.
7.  if all tests pass, the test was successful.   ETD keys for all instances (Dev/QA/Prod) are in 1pass. 

PR to place ETD public keys in the corresponding instance in the DAIS-IF repo is at: https://github.huit.harvard.edu/LTS/DAIS-IF/pull/101

# Test coverage
Yes/No: Are changes in this pull-request covered by:
- unit tests? y
- integration tests? n

# Interested parties
Tag (@ mention) interested parties: @awoods
